File: Modificador de 4 lenguaje/js/claseabstractas/Ingeniero.js
const Empleado = require("./Empleado");

class Ingeniero extends Empleado {
    constructor(nombre, horas, tarifa) {
        super(nombre);
        this.horas = horas;
        this.tarifa = tarifa;
    }

    calcularSalario() {
        return this.horas * this.tarifa;
    }
}

module.exports = Ingeniero;

File: Modificador de 4 lenguaje/C#/ClasesAbstractas/ClasesAbstractas/Ingeniero.cs
namespace EjemploConstructores.ClasesAbstractas
{
    class Ingeniero : Empleado
    {
        // Constructor sin parámetros
        public Ingeniero() { Nombre = "Ingeniero"; }

        // Constructor de copia
        public Ingeniero(Ingeniero otro) { Nombre = otro.Nombre; }

        public override void Trabajar() => System.Console.WriteLine($"{Nombre} está trabajando.");
    }
}
